compiler/rustc_mir_transform/src/gvn.rs

```diff
@@ -418,9 +421,7 @@ impl<'body, 'tcx> VnState<'body, 'tcx> {
                         self.ecx.copy_op(op, &field_dest).ok()?;
                     }
                     self.ecx.write_discriminant(variant.unwrap_or(FIRST_VARIANT), &dest).ok()?;
-                    self.ecx
-                        .alloc_mark_immutable(dest.ptr().provenance.unwrap().alloc_id())
-                        .ok()?;
```

Comment:
Why do we stop marking as immutable?

Reply:
Correct me if I'm wrong but I think ` let dest = dest.map_provenance(|prov| prov.as_immutable());` in the line below could serve the same purpose.